### ISSUE TITLE
Unable to use the new DNS endpoint exclusively.

### ISSUE THREAD

**Issue by TheKangaroo** [terraform-google-modules/terraform-google-kubernetes-engine#2216]
### TL;DR

My goal is to use only the DNS endpoint on my GKE clusters:
![image](https://github.com/user-attachments/assets/5db28e87-2e2f-481c-8851-ee92a0bb9aac)

This is not possible at the moment because I need to set:

```
  enable_private_endpoint       = true
  deploy_using_private_endpoint = true
```

for the DNS endpoint to be enabled. However, `enable_private_endpoint` will create an empty `master_authorized_networks_config {}` block in [this line](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/9a86b4a894dd93364550fd4194e6f2848d30e8f9/modules/beta-private-cluster-update-variant/cluster.tf#L217), and applying this will re-enable access via IPv4.
![image](https://github.com/user-attachments/assets/0c2928ac-e005-4076-b41f-b129a4513ed5)

I created [a PR to fix this](#2217), but I'm not 100% sure if it breaks anything unrelated to DNS endpoints.

Another thing I observed while experimenting with this setting is that removing `master_authorized_networks_config {}` won't actually change my cluster to disable IPv4 access. The code seems to work for both cases: with `master_authorized_networks_config {}` and IPv4 enabled, and omitted with IPv4 disabled. Maybe this is just a limitation in the API, where changing `master_authorized_networks_config {}` to `(empty)` does not trigger an update. 🤔

### Terraform Configuration

```hcl
n.a.
```

### Terraform Version

```sh
tofu version                                                                                          app-publisher-dev-7ed19f10be
OpenTofu v1.8.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/google v4.84.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.23.0
+ provider registry.terraform.io/hashicorp/random v3.5.1
```

**Comment by apeabody** [comment]
Thanks @TheKangaroo - Interesting, it looks like the presence of the (empty) `master_authorized_networks_config` block is actually being used as a substitute for `enabled`: https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/container/resource_container_cluster.go#L5035

**Comment by apeabody** [comment]
Interesting - So currently the module's `enable_private_endpoint` actuates the provider's `private_cluster_config.enable_private_endpoint` which actuates the API's `PrivateClusterConfig.enablePrivateEndpoint` which is [deprecated](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.PrivateClusterConfig).  The recommendation is to use `ControlPlaneEndpointsConfig.IPEndpointsConfig.enable_public_endpoint`, however `ControlPlaneEndpointsConfig.IPEndpointsConfig.enabled` is currently [hardcoded to true](https://github.com/hashicorp/terraform-provider-google/blame/v6.14.1/google/services/container/resource_container_cluster.go#L5151) in the Provider.

As `masterAuthorizedNetworksConfig` is part of `ControlPlaneEndpointsConfig.IPEndpointsConfig`, if `masterAuthorizedNetworksConfig` is created using the Provider, it is likely resulting in the side-effect of `ControlPlaneEndpointsConfig.IPEndpointsConfig.enabled = true`.

**Comment by apeabody** [comment]
Here is a PR to add Provider support  for `ControlPlaneEndpointsConfig.IPEndpointsConfig.enabled`: https://github.com/hashicorp/terraform-provider-google/issues/20369

**Comment by TheKangaroo** [comment]
Ah, I think I understand now. Thanks for reviewing the current implementation and providing the explanation. Since we can already use DNS endpoints and disabling IP endpoints would just be an added benefit, I'm fine with waiting for https://github.com/hashicorp/terraform-provider-google/issues/20369.

**Comment by asafhm** [comment]
> Here is a PR to add Provider support for `ControlPlaneEndpointsConfig.IPEndpointsConfig.enabled`: [hashicorp/terraform-provider-google#20369](https://github.com/hashicorp/terraform-provider-google/issues/20369)

That provider's PR is merged. Can we fix this in the module?

**Comment by TheKangaroo** [comment]
I haven't been able to find the time to work on the PR since then, but I will do so as soon as possible.
